src/abstractdataframe/show.jl

```diff
@@ -573,61 +573,203 @@ function _show(io::IO,
                df::AbstractDataFrame;
                allrows::Bool = !get(io, :limit, false),
                allcols::Bool = !get(io, :limit, false),
-               splitcols = get(io, :limit, false),
                rowlabel::Symbol = :Row,
                summary::Bool = true,
                eltypes::Bool = true,
-               rowid=nothing,
-               truncstring::Int)
+               rowid = nothing,
+               truncate::Int = 32,
+               kwargs...)
+
     _check_consistency(df)
 
-    # we will pass around this buffer to avoid its reallocation in ourstrwidth
-    buffer = IOBuffer(Vector{UInt8}(undef, 80), read=true, write=true)
+    aux = names(df)
+    names_len = textwidth.(aux)
+    maxwidth = max.(9, names_len)
+    names_mat = permutedims(aux)
+    types = eltype.(eachcol(df))
+
+    # NOTE: If we reuse `types` here, the time to print the first table is 2x more.
+    # This should be something related to type inference.
+    types_str = permutedims(compacttype.(eltype.(eachcol(df)), maxwidth))
+
+    if allcols && allrows
+        crop = :none
+    elseif allcols
+        crop = :vertical
+    elseif allrows
+        crop = :horizontal
+    else
+        crop = :both
+    end
+
+    compact_printing::Bool = get(io, :compact, true)
 
-    nrows = size(df, 1)
-    if rowid !== nothing
-        if size(df, 2) == 0
-            rowid = nothing
-        elseif nrows != 1
-            throw(ArgumentError("rowid may be passed only with a single row data frame"))
+    num_rows, num_cols = size(df)
+
+    # By default, we align the columns to the left unless they are numbers,
+    # which is checked in the following.
+    alignment = [:l for i = 1:num_cols]
+
+    # This vector stores the column indices that are only floats. In this case,
+    # the printed numbers will be aligned on the decimal point.
+    float_cols = Int[]
+
+    # These vectors contain the number of the row and the padding that must be
+    # applied so that the float number is aligned with on the decimal point
+    indices = Vector{Int}[]
+    padding = Vector{Int}[]
+
+    # If the screen is limited, we do not need to process all the numbers.
+    dsize = displaysize(io)
+
+    if !allcols
+        # Given the spacing, there is no way to fit more than W/9 rows of
+        # floating numbers in the screen, where W is the display width.
+        Δc = clamp(div(dsize[2], 9), 0, num_cols)
+    else
+        Δc = num_cols
+    end
+
+    if !allrows
+        # Get the maximum number of lines that we can display given the screen size.
+        Δr = clamp(dsize[1] - 4, 0, num_rows)
+    else
+        Δr = num_rows
+    end
+
+    Δr_lim = cld(Δr, 2)
+
+    for i = 1:Δc
+        type_i = nonmissingtype(types[i])
+
+        if type_i <: AbstractFloat
+            # Do not align the numbers if there are more than 500 rows.
+            if Δr ≥ 500
```

Comment:
Is there any specific reason to use 500 rather than a higher limit? I'd tend to prefer aligning numbers as long as it's not pathologically slow, as people will be surprised if they get a different printing. For example, 5000 or 10000 sounds safer to me.
```suggestion
            # Do not align the numbers if there are more than 5000 rows.
            if Δr ≥ 5000
```

Reply:
> I'd tend to prefer aligning numbers as long as it's not pathologically slow

I think we can use 5000 if PrettyTables.jl considers here only shown columns (so that it we have a `10000 x 10000` data frame we do not wait extremely long). But I assume this is the case - right?

Reply:
The problem with 5000 is that a `5000 x 5000` data frame composed only of float columns take ages to print if `allrows = true, allcols = true`. However, if I crop columns in a terminal with 252 columns, then it takes less than 1s.